dags/aps/parser.py

```diff
@@ -64,11 +65,11 @@ def __init__(self) -> None:
             NestedValueExtractor(
                 "copyright_statement", json_path="rights.rightsStatement"
             ),
-            NestedValueExtractor(
+            CustomExtractor(
                 "license",
-                json_path="rights.licenses",
+                extraction_function=lambda x: self._get_licenses(x['rights']['licenses'][0])
```

Comment:
does it always have licenses? check what we have also for Springer for licenses.

Reply:
Springer parses XML, so it's a string. For APS it's JSON, and it's always in the array.
However, I can add catch/except block

Reply:
It doesn't matter. We can modify the license function to accept parameters that can be used for both, anyway for both publishers, we should return a JSON

Reply:
The logic is pretty similar https://github.com/cern-sis/workflows/blob/main/dags/springer/parser.py#L240-L266

Reply:
but I need to extract it from JSON, not from XML, in the end, I cannot reuse it.
The license mappings I took from scoap3/extractors: https://github.com/SCOAP3/hepcrawl/blob/c10a838e94b6457356aabb9bb370eef35ec2dc60/hepcrawl/mappings.py#L189-L193

Reply:
We can move the extraction parts on the publisher's parser and have a common function to return the licenses